### PR TITLE
Upgraded the meshkit error

### LINKED ISSUE #198: Upgrade meshkit errors
**Enhancement Description**
Port meshkit errors to the new error utility function. Instructions covered here, layer5io/meshkit#48

### PR DESCRIPTION
**Pull request by sayantan1413**
Signed-off-by: sayantan1413 <sayantanbose2001@gmail.com>

Upgraded the meshkit error.

Issue fixes #198

### CODE REVIEW

File: internal/config/error.go [hunk 1]
```diff
@@ -15,15 +15,13 @@
 package config
 
 import (
-	"fmt"
-
 	"github.com/layer5io/meshkit/errors"
 )
 
 const (
 	ErrEmptyConfigCode           = "11300"
```
[the remark this answers is not in the document]
Comment (mgfeller) [reply internal/config/error.go:22]: and would be good to replace its `error.NewDefault`

File: linkerd/error.go [hunk 2]
```diff
@@ -2,83 +2,83 @@
...
+	ErrSampleAppCode = "1010"
 	// ErrCustomOperationCode is the error code for ErrCustomOperation
-	ErrCustomOperationCode = "linkerd_test_code"
+	ErrCustomOperationCode = "1011"
+	// ErrOpInvalidCode is the error code for ErrOpInvalid
+	ErrOpInvalidCode = "1012"
 
 	// ErrOpInvalid is the error for invalid operation
-	ErrOpInvalid = errors.NewDefault(errors.ErrOpInvalid, "Invalid operation")
+	ErrOpInvalid = errors.New(ErrOpInvalidCode, errors.Alert, []string{"Invalid Operation"}, []string{}, []string{}, []string{})
```
Comment (mgfeller) [inline linkerd/error.go:33]: capitalization?

**Review by mgfeller — CHANGES_REQUESTED** [review]
Nice :smile: 

I wonder, in general, how do we handle capitalization, e.g. in error messages? first letter capitalized or not? capitalization of mesh name, i.e. Linkerd instead of linkerd?

File: internal/config/error.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (sayantan1413) [reply internal/config/error.go:22]: Ahh yeah @mgfeller I somehow overlooked this. 😬

File: linkerd/error.go (hunk 2 above)
Comment (sayantan1413) [reply linkerd/error.go:33]: Oh yes I need to change the `Operation` to `operation`.

**Comment by sayantan1413** [comment]
> Nice 😄
> 
> I wonder, in general, how do we handle capitalization, e.g. in error messages? first letter capitalized or not? capitalization of mesh name, i.e. Linkerd instead of linkerd?

Yes @mgfeller I think we can follow the order like capitalizing the first letter of the each error and only capitalize the mesh names if it is present in between the error string.

**Comment by sayantan1413** [comment]
Hey @mgfeller can you review it once more ? I did the changes that you requested.

**Review by mgfeller — CHANGES_REQUESTED** [review]
@sayantan1413 there is a duplicate code: "1002", otherwise :+1: 
In the next version of the utility, this will result in a non-zero exit code, to make it easier to notice.
Any other improvements you see?

**Comment by sayantan1413** [comment]
Hey @mgfeller I did the change you can review it once more. I am also thinking of a way to not duplicate the error codes.

**Review by mgfeller — CHANGES_REQUESTED** [review]
@sayantan1413 hmm did you change the codes manually? The output files are not in sync with the code. If codes are changed manually, `update` or `analyze` should be executed to sync the output files. `next_error_code` in  `component_info.json` also needs to be updated manually (as an improvement, I'll implement a check/correction for that too). Alternatively, `update`can be run with the `--force` flag to renumber all codes (setting `next_error_code` to the first on). 

Duplicate codes are on purpose not fixed automagically, as it is not really possible to find out which one is the faulty one, IMHO.

Not sure if I can push to your branch, but if so, I can if you like.

**Comment by sayantan1413** [comment]
Hey @mgfeller please check now I think the duplicate codes are removed now and the `next_error_code` is updated with the latest value.

**Comment by sayantan1413** [comment]
This time I didn't change the code manually rather I placed a placedholder like `replace_me` and then after running the `make error` command everything got updated in the JSON file automatically.

**Comment by mgfeller** [comment]
@sayantan1413 something is not working, I need to check that out. Running update twice produces the desired result, but is not optimal. I have an idea why it's not working. 
If it is ok, I can try to push to your repo (not sure whether and where I can see whether the option to let maintainers to push to your branch is active or not).

**Comment by sayantan1413** [comment]
Yeah @mgfeller I think you can push your changes in my branch. Though I don't know how exactly that works, I will give it a check and inform you if I get anything.

**Comment by mgfeller** [comment]
Thanks @sayantan1413, sorry for the inconvenience. It'll be tomorrow though, PC is shut down for today.

**Comment by sayantan1413** [comment]
Yeah @mgfeller sure we can do this tomorrow.😃

**Comment by mgfeller** [comment]
@sayantan1413 I was actually able to push to this PR :smile:  Could you have a look, then we can close this if all looks ok.
And I'll create a new issue to improve and fix the error util tool.

**Comment by sayantan1413** [comment]
Hey @mgfeller this PR now look good to me. I think you can now close it.

**Comment by sayantan1413** [comment]
Hey @mgfeller could you please tell me that how did you replace all the duplicates ?

**Comment by mgfeller** [comment]
I just ran `make error` once more. This should really not be necessary (and wasn't intended), the tool should handle that - I thought it would (and that I tested that) but obviously missed it. I'll create an issue and fix it.

**Comment by leecalcote** [comment]
Thank you, @mgfeller. I, too, thought idempotency in the Error Code Utility already vetted and good to go.

**Comment by sayantan1413** [comment]
Okay @mgfeller. Thank you @mgfeller I had a good learning session with you. I really enjoyed creating this PR with a quality learning session. 😃 I think this PR is good to go now.

**Comment by mgfeller** [comment]
> Thank you, @mgfeller. I, too, thought idempotency in the Error Code Utility already vetted and good to go.

Ah, one of the important magic words in a world of distributed microservices :smile: also applicable in other contexts.

**Comment by mgfeller** [comment]
> Okay @mgfeller. Thank you @mgfeller I had a good learning session with you. I really enjoyed creating this PR with a quality learning session. smiley I think this PR is good to go now.

Thank you @sayantan1413, likewise!

**Review by mgfeller — APPROVED** [review]
Good to go!

**Pull request merged**